[ macOS ] fast/speechrecognition/start-recognition-after-denied-gum.html is a flaky Timeout

youennf opened the pull request:
#### ad13d163120d5e9bba5017c698e95994f6b88e28
<pre>
[ macOS ] fast/speechrecognition/start-recognition-after-denied-gum.html is a flaky Timeout
<a href="https://rdar.apple.com/123722096">rdar://123722096</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=270194">https://bugs.webkit.org/show_bug.cgi?id=270194</a>

Reviewed by Eric Carlson.

Make sure to not GC SpeechRecognition if it can fire events and has event listeners.

* LayoutTests/platform/mac-wk2/TestExpectations:
* Source/WebCore/Modules/speech/SpeechRecognition.cpp:
(WebCore::SpeechRecognition::virtualHasPendingActivity const):
* Source/WebCore/Modules/speech/SpeechRecognition.h:

Canonical link: <a href="https://commits.webkit.org/275969@main">https://commits.webkit.org/275969@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/03d70a00b16cada1ad48ba623e32320f1ce1a8ac

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/43415 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/22446 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/45826 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/46050 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/39541 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/45719 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/26220 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/19864 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/35889 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/43989 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/19475 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/37393 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/16866 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/17050 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/38457 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/1471 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/39608 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/38767 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/47594 "Built successfully") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/18374 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/15081 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/42672 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/19867 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/41336 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9662 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/20047 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/19498 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

szewai commented on Source/WebCore/Modules/speech/SpeechRecognition.cpp:
```diff
@@ -216,4 +216,9 @@ SpeechRecognition::~SpeechRecognition()
 
+bool SpeechRecognition::virtualHasPendingActivity() const
+{
+    return (m_state == State::Starting || m_state == State::Running) && hasEventListeners();
```
This should probably be `m_state != State::Inactive && hasEventListeners()`

youennf replied:
Right, done

youennf committed:
[ macOS ] fast/speechrecognition/start-recognition-after-denied-gum.html is a flaky Timeout
rdar://123722096
https://bugs.webkit.org/show_bug.cgi?id=270194

Reviewed by Eric Carlson.

Make sure to not GC SpeechRecognition if it can fire events and has event listeners.

* LayoutTests/platform/mac-wk2/TestExpectations:
* Source/WebCore/Modules/speech/SpeechRecognition.cpp:
(WebCore::SpeechRecognition::virtualHasPendingActivity const):
* Source/WebCore/Modules/speech/SpeechRecognition.h:

Canonical link: https://commits.webkit.org/275969@main
